elastic/kibana
# [Synthetics] Add overview grid

## Summary
Resolves https://github.com/elastic/kibana/issues/135147

Adds basic Overview monitor grid that displays a Metric chart for each monitor/location combination.

This metric displays the monitor duration trend for the last 50 test runs.
Disabled monitors
<img width="1537" alt="Screen Shot 2022-07-14 at 9 36 28 AM" src="https://user-images.githubusercontent.com/11356435/178998696-cb9ef4c8-fe38-4435-9a60-9a72957dd509.png">
Up and down monitors with limited data (no trend line)
<img width="1538" alt="Screen Shot 2022-07-14 at 9 36 06 AM" src="https://user-images.githubusercontent.com/11356435/178998698-2cf99180-6d7f-45e7-a234-94337380a4f5.png">

Up and down monitors with data
<img width="1539" alt="Screen Shot 2022-07-14 at 9 54 14 AM" src="https://user-images.githubusercontent.com/11356435/178999280-c4d1cb80-3e4d-471c-b742-ba9db91933e6.png">

Monitors without data
<img width="1540" alt="Screen Shot 2022-07-14 at 9 35 16 AM" src="https://user-images.githubusercontent.com/11356435/178998700-ae16ede2-bbb4-4600-b672-b909aa16dad8.png">


Testing
--
This PR is deployed in cloud at: https://deploy-kibana-qxqfu-custom.kb.us-west2.gcp.elastic-cloud.com:9243
Credentials: https://p.elstc.co/paste/Nn3sWkQV#tZkz8ypUDPPYeEeRPo1jzv9df8acyrWOv45wUO6O2nu

1. Navigate to `/app/synthetics`.
2. Add your first monitor, ideally selecting all locations
3. Navigate back to `/app/synthetics`.
4. Before data is available, you should see green grid items displaying N/A as the duration.
5. Wait 30 seconds. The overview page is designed to reload every 30 seconds
6. You should start to see data coming in, with grid items filling in the computed monitor duration
7. The trend line takes a few checks to start populating correctly. Wait a few test checks to see the trend lines in the chart. You may want to edit your monitor to run every 1 or 2 minutes.
8. Monitors with current successful check should be green, while failing monitors should be red
9. There is currently a Synthetics service bug preventing data for the Europe - United Kingdom location from indexing data correctly. This grid item will always display N/A with no data
10. Navigate to `/app/synthetics/monitors` in a new tab. Edit your first monitor to ensure that it fails.
11. Navigate back to your old tab. Wait for data to start streaming in. You should see the green metric item change to red. 
12. Click on the monitors tab, disable your monitor. Click back on the overview tab.
13. You should see the metric items are now greyed out, indicating they are disabled.
14. Add another 1 or 2 monitors with all locations
15. Navigate back to `/app/synthetics`. Adjust the pagination settings and navigate back and forth with pagination.
